utils/src/ast-grep/shebang.ts

```diff
@@ -0,0 +1,75 @@
+import type { SgRoot, Edit } from '@codemod.com/jssg-types/main';
+
+/**
+ * Get the shebang line from the root.
+ * @param root The root node to search.
+ * @returns The shebang line if found, otherwise null.
+ */
+export const getShebang = (root: SgRoot) =>
+	root.root().find({
+		rule: {
+			kind: 'hash_bang_line',
+			regex: '\\bnode(\\.exe)?\\b',
+			not: {
+				// tree-sitter wrap hash bang in Error node
+				// when it's not in the top of program node
+				inside: {
+					kind: 'ERROR',
+				},
+			},
+		},
+	});
+
+/**
+ * Replace Node.js arguments in the shebang line.
+ * @param root The root node to search.
+ * @param argsToValues The mapping of argument names to their new values.
+ * @param edits The list of edits to apply.
+ * @returns The updated shebang line if any replacements were made, otherwise null.
+ */
+/**
+ * Replace Node.js arguments in the shebang line and return edits.
+ * @param root The root node to search.
+ * @param argsToValues The mapping of argument names to their new values.
+ * @returns Array of edits to apply (empty if none).
+ */
+export const replaceNodeJsArgs = (
+	root: SgRoot,
+	argsToValues: Record<string, string>,
+): Edit[] => {
+	const edits: Edit[] = [];
+	const shebang = getShebang(root);
+
+	if (!shebang) return edits;
+
+	const text = shebang.text();
+	// Find the "node" argument in the shebang
+	const nodeMatch = text.match(/\bnode(\.exe)?\b/);
+
+	if (!nodeMatch) return edits;
+
+	// We only touch to something after node because before it's env thing
+	const nodeIdx = nodeMatch.index! + nodeMatch[0].length;
+	const beforeNode = text.slice(0, nodeIdx);
+	let afterNode = text.slice(nodeIdx);
+
+	for (const argC of Object.keys(argsToValues)) {
+		// Escape special regex characters in arg
+		const esc = argC.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
```

Comment:
nit: I think you can cache this regex (move it out of the function) so it isn't re-constructed every loop iteration (and even function execution).